### PR TITLE
Github action for build and push a docker image to cr

### PR DESCRIPTION
**Pull request by Tansito**
First export to a reusable github action.

This github action will manage:
- build of the docker image
- push of the docker image to the container registry
- clean of docker image older versions in the container registry in case they exist

### CODE REVIEW

File: .github/workflows/build-publish-docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      IBMCLOUD_ACCOUNT:
+        required: true
+      IBMCLOUD_API_KEY:
+        required: true
+
+jobs:
+  build-publish-docker-image:
+    runs-on: ubuntu-latest
+    env:
+      DOCKER_IMAGE_TAG: icr.io/community-digital/${{ inputs.DOCKER_IMAGE_NAME }}:${{ inputs.PR_TAG }}
```
Comment (abdonrd) [inline .github/workflows/build-publish-docker-image.yml:28]: We should avoid hard-code `community-digital`, right?

Comment (Tansito) [reply .github/workflows/build-publish-docker-image.yml:28]: good catch, I agree

**Review by abdonrd — APPROVED** [review]
It seems like a good start!

**Comment by vabarbosa** [comment]
@Tansito @abdonrd i know i am late to the party, but i had a question. 

since this repo is intended for a collection of workflows and actions, does it make sense to move the action into a specific subfolder  (e.g., `code-engine-preview/action.yml`) instead of where it currently is (`.github/workflows/code-engine-preview.yml`)?

that way as more actions are created and shared they can each be in their own directory.

**Comment by Tansito** [comment]
The reason because I created that structure is because from the documentation ([here](https://docs.github.com/en/actions/learn-github-actions/reusing-workflows#creating-a-reusable-workflow)) it seems that it's the only available way to configure reusable workflows (If I understood well). And it seems that they don't accept subfolders either.

My next step now that It seems that the workflow is working is try to separate it in more little workflows to be able to reuse them.

**Comment by vabarbosa** [comment]
> The reason because I created that structure is because from the documentation ([here](https://docs.github.com/en/actions/learn-github-actions/reusing-workflows#creating-a-reusable-workflow)) it seems that it's the only available way to configure reusable workflows (If I understood well). And it seems that they don't accept subfolders either.
> 
> My next step now that It seems that the workflow is working is try to separate it in more little workflows to be able to reuse them.
> 
> And never it's too late to enter into the party 😄

oh ok. i didnt realize it was a requirement. good to know. thanks!

**Comment by Tansito** [comment]
Yeah, another very interesting limitation is that you can't call another reusable workflow from a reusable workflow. It has sense once time you think about it but just something to take into account too.

**Comment by eddybrando** [comment]
Thank you very much for this PR, @Tansito! I like where this is going 🙂.

**Pull request merged**